ts/packages/anchor/src/provider.ts

```diff
@@ -367,32 +377,71 @@ export interface Wallet {
 async function sendAndConfirmRawTransaction(
   connection: Connection,
   rawTransaction: Buffer | Uint8Array,
-  options?: ConfirmOptions
+  options?: ConfirmOptions,
+  blockhash?: BlockhashWithExpiryBlockHeight
 ): Promise<TransactionSignature> {
-  const sendOptions = options && {
-    skipPreflight: options.skipPreflight,
-    preflightCommitment: options.preflightCommitment || options.commitment,
+  const sendOptions: SendOptions = {
+    skipPreflight:
+      options?.skipPreflight !== undefined ? options.skipPreflight : true,
+    preflightCommitment:
+      options?.preflightCommitment || options?.commitment || "confirmed",
+    minContextSlot: options?.minContextSlot,
+    maxRetries: options?.maxRetries || 0,
   };
 
-  const signature = await connection.sendRawTransaction(
-    rawTransaction,
-    sendOptions
-  );
-
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment
-    )
-  ).value;
-
-  if (status.err) {
-    throw new ConfirmError(
-      `Raw transaction ${signature} failed (${JSON.stringify(status)})`
-    );
+  let status: SignatureResult;
+
+  const startTime = Date.now();
+  while (Date.now() - startTime < 60_000) {
+    try {
+      const signature = await connection.sendRawTransaction(
+        rawTransaction,
+        sendOptions
+      );
+
+      if (blockhash) {
+        if (sendOptions.maxRetries === 0) {
+          const abortSignal = AbortSignal.timeout(15_000);
+          status = (
+            await connection.confirmTransaction(
+              { abortSignal, signature, ...blockhash },
+              sendOptions.preflightCommitment
+            )
+          ).value;
+          abortSignal.removeEventListener("abort", () => {});
```

Comment:
I don't think this would do anything useful since you're creating a new listener.